usecases/schema/parser.go

```diff
@@ -476,3 +490,65 @@ func validateShardingConfig(current, update *models.Class, mtEnabled bool) error
 	}
 	return nil
 }
+
+// This is used to validate whether the schema of a class to be updated is attempting to mutate any of the immutable fields of a models.Property
+// For now, only Description is allowed to be mutated so this field is not included in this struct.
+type immutableProperty struct {
```

Comment:
I think I would explicitly list the fields that are mutable if that is possible 🤔 Most fields we add are immutable and I am sure they will not be added here

Reply:
I've flipped the logic around to convert `models.Property` and `models.NestedProperty` into maps that lack the `description` key. These maps are then compared. This way, future additions to these structs won't need to be propagated here unless they're mutable